src/domain/entities/figma-design-identifier.ts

```diff
@@ -33,18 +33,17 @@ export class FigmaDesignIdentifier {
 	 *
 	 * The method parses the URL and extracts data required for building {@link FigmaDesignIdentifier}.
 	 */
-	static fromFigmaDesignUrl = (url: string): FigmaDesignIdentifier => {
-		const parsedUrl = new URL(url);
-
-		const pathComponents = parsedUrl.pathname.split('/');
+	static fromFigmaDesignUrl = (url: URL): FigmaDesignIdentifier => {
```

Comment:
**note:** Changes the contract to accept `URL` so the function should not worry about invalid URLs and related error handling.